crates/ra_ide_api/tests/test/main.rs

```diff
@@ -167,15 +168,47 @@ fn test_rename_for_mut_param() {
     );
 }
 
+#[test]
+fn test_rename_mod() {
+    let (analysis, position) = analysis_and_position(
+        "
+        //- /bar.rs
+        mod fo<|>o;
+        //- /bar/foo.rs
+        // emtpy
+    ",
+    );
+    let new_name = "foo2";
+    let source_change = analysis.rename(position, new_name).unwrap();
+    assert_debug_snapshot_matches!("rename_mod", &source_change);
+}
+
+#[test]
+fn test_rename_mod_in_dir() {
+    let (analysis, position) = analysis_and_position(
+        "
+        //- /lib.rs
+        mod fo<|>o;
+        //- /foo/mod.rs
+        // emtpy
+    ",
+    );
+    let new_name = "foo2";
+    let source_change = analysis.rename(position, new_name).unwrap();
+    assert_debug_snapshot_matches!("rename_mod_in_dir", &source_change);
+}
+
```

Comment:
While we are at it, could we move these and preexisting rename tests to that `rename.rs` file? That is, let's make rename tests unit-tests.

Reply:
moving to inline tests need snapshot to be here ` Snapshot: /Users/feichao/Develop/rust-analyzer/crates/ra_ide_api/src/snapshots/tests__rename_mod.snap`

Reply:
Yep, it's the right place.

Reply:
ok